docs/content/concepts/io-management/io-managers.mdx

```diff
@@ -278,6 +337,88 @@ class MyPartitionedIOManager(IOManager):
         return read_csv(self.path_for_partition(context.asset_partition_key))
 ```
 
+### Writing a per-input IO Manager
+
+In some cases you may find that you need to load an input in a way other than the `load_input` function of the corresponding output's IO manager. For example, let's say Team A has an op that returns an output as a Pandas DataFrame and specifies an IO manager that knows how to store and load Pandas DataFrames. Your team is interested in using this output for a new op, but you are required to use PySpark to analyze the data. Unfortunately, you don't have permission to modify Team A's IO manager to support this case. Instead, you can specify an input manager on your op that will override some of the behavior of Team A's IO manager.
+
+Since the method for loading an input is directly affected by the way the corresponding output was stored, we recommend defining your input managers as subclasses of existing IO managers and just updating the `load_input` method. In this example, we load an input as a NumPy array rather than a Pandas DataFrame by writing the following:
+
+```python file=/concepts/io_management/input_managers.py startafter=start_plain_input_manager endbefore=end_plain_input_manager
+# in this case PandasIOManager is an existing IO Manager
+class MyNumpyLoader(PandasIOManager):
```

Comment:
Would the `@input_manager` decorator make sense in this case?

Reply:
i think in this first version it could be a plain input manager, but in the better version of the example it has to be an IO Manager because it shares the `_get_path` function defined by the PandasIOManager. I think the example is more streamlined if we just start off with an IO manager instead of going from input manager to io manager